influxql/select.go

```diff
@@ -355,6 +354,25 @@ func buildRHSTransformIterator(lhs Iterator, rhs Literal, op Token, ic IteratorC
 				return p
 			},
 		}, nil
+	case func(int64, int64) int64:
```

Comment:
I agree that it won't operate correctly right now. There's definitely some casting support that needs to happen but it's complicated enough that I'd rather push it to another PR. We also need to cast iterators that come from multiple shards that have different data types (e.g. when someone changes the data type they're inserting as).